pkg/extensions/search/utils/db.go

```
@@ -0,0 +1,229 @@
+	if uerr != nil {
+		fmt.Println("Unable to Update Name Package Bucket")
+
+		return false
+	}
+
+	uerr = updateNVDPkg(NameverDB, mapList[2], db)
+	if uerr != nil {
+		fmt.Println("Unable to Update Name-Version Package Bucket")
+
+		return false
+	}
+
+	return err == nil
+}
+
+/*UpdateNVD ... Updating the NVD database */
+func updateNVDPkg(name string, pkgList map[string][]CVEId, db *bolt.DB) error {
+	var dbcveidlist []CVEId
+
+	err := db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(name))
+		for pkg, cveidlist := range pkgList {
+			v := b.Get([]byte(pkg))
+			if v == nil {
+				encode, _ := json.Marshal(cveidlist)
+
+				err := b.Put([]byte(pkg), encode)
+				if err != nil {
+					fmt.Println("Unable to Insert Data from PkgVendor Bucket")
+
+					return err
+				}
+			} else {
+				err := json.Unmarshal(v, &dbcveidlist)
+				if err != nil {
+					fmt.Println("Unable to Unmarshal Data from PkgVendor Bucket")
+
+					return err
+				}
+
+				cveidlist = append(cveidlist, dbcveidlist...)
+
+				encode, _ := json.Marshal(cveidlist)
+
+				err = b.Put([]byte(pkg), encode)
+				if err != nil {
+					fmt.Println("Unable to Insert Data from PkgVendor Bucket")
+
+					return err
+				}
+			}
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+/* Updating the NVD Meta Database */
+func updateNVDMeta(filepath string, hashcode string, db *bolt.DB) bool {
+	err := db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte("NvdMeta"))
+
+		err := b.Put([]byte(filepath), []byte(hashcode))
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	return err == nil
+}
+
+// Method to check if file content is already present in DB
+func isPresent(filename string, hashcode string, db *bolt.DB) bool {
+	var v []byte
+
+	err := db.View(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(NvdmetaDB))
+
+		v = b.Get([]byte(filename))
+
+		return nil
+	})
+	if err != nil {
+		fmt.Println("Not able to search")
+		fmt.Println(err)
```

Looks like all these functions should return an error.

---

Ok, I will change all method return type as error and update the code accordingly.